### ISSUE TITLE
Crash on iOS (when handling a deep link)

### ISSUE THREAD

**Issue by optikalefx** [BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK#125]
I get the following crash in Xcode when hitting `myapp://`

```
2016-04-25 23:18:57.758 FactSumo[3217:1602016] returning data to js interface..
2016-04-25 23:18:57.758 FactSumo[3217:1602016] Sending to JS: {
    "+clicked_branch_link" = 0;
    "+is_first_session" = 0;
}
2016-04-25 23:19:00.352 FactSumo[3217:1602016] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Invalid type in JSON write (NSURL)'
*** First throw call stack:
(0x182e1ee38 0x182483f80 0x182e1ed80 0x18386b1d4 0x18386d364 0x182d6a50c 0x182d581b0 0x18386caf0 0x18386b0a0 0x18386aed0 0x18386bf2c 0x1000ee1bc 0x182dc0eac 0x182dc06cc 0x182dc044c 0x182e29494 0x182cfe788 0x1000ef0c8 0x188219990 0x188219400 0x188221e58 0x188228594 0x188227990 0x18821161c 0x1882112a4 0x1847bf7ac 0x1847bf618 0x1847bf9c8 0x182dd5124 0x182dd4bb8 0x182dd28b8 0x182cfcd10 0x1845e4088 0x187fd1f70 0x100084cc4 0x18289a8b8)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

**Comment by aaustin** [comment]
Hey @optikalefx - I believe I've resolved this in my open PR https://github.com/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/pull/120. I'm waiting on some tests to confirm before I merge it in.

**Comment by barocsi** [comment]
https://github.com/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/issues/123

**Comment by aaustin** [comment]
Knocked this one out in 2.0.12
